python/mxnet/test_utils.py

```diff
@@ -203,55 +266,98 @@ def rand_sparse_ndarray(shape, stype, density=None, distribution="uniform", dtyp
+            val.fill(data_init)
+        if modifier_func is not None:
+            val = assign_each(val, modifier_func)
+
         arr = mx.nd.row_sparse_array(val, indices, shape, indices_type=np.int64, dtype=dtype)
         return arr, (val, indices)
     elif stype == 'csr':
         assert len(shape) == 2
         if distribution == "uniform":
-            csr = _get_uniform_dataset_csr(shape[0], shape[1], density, dtype=dtype)
+            csr = _get_uniform_dataset_csr(shape[0], shape[1], density,
+                                           data_init=data_init,
+                                           shuffle_csr_indices=shuffle_csr_indices, dtype=dtype)
             return csr, (csr.indptr, csr.indices, csr.data)
         elif distribution == "powerlaw":
-            csr = _get_powerlaw_dataset_csr(shape[0], shape[1], density, dtype=dtype)
+            csr = _get_powerlaw_dataset_csr(shape[0], shape[1], density=density, dtype=dtype)
             return csr, (csr.indptr, csr.indices, csr.data)
         else:
             assert(False), "Distribution not supported: %s" % (distribution)
     else:
         assert(False), "unknown storage type"
 
-
-def rand_ndarray(shape, stype, density=None, dtype=None):
+def rand_ndarray(shape, stype, density=None, dtype=None, modifier_func=None):
     if stype == 'default':
         arr = mx.nd.array(random_arrays(shape), dtype=dtype)
     else:
-        arr, _ = rand_sparse_ndarray(shape, stype, density=density, dtype=dtype)
+        arr, _ = rand_sparse_ndarray(shape, stype, density=density,
+                                     modifier_func=modifier_func, dtype=dtype)
     return arr
 
 
+def create_sparse_array(shape, stype, data_init=None, rsp_indices=None,
+                        dtype=None, modifier_func=None, density=.5):
+    if stype == 'row_sparse':
+        if rsp_indices is not None:
+            arr_indices = np.asarray(rsp_indices)
+            arr_indices.sort()
+        else:
+            arr_indices = None
+        arr_data, (_, _) = rand_sparse_ndarray(shape, stype,
+                                               density=density,
+                                               data_init=data_init,
+                                               rsp_indices=arr_indices,
+                                               dtype=dtype,
+                                               modifier_func=modifier_func)
+    elif stype == 'csr':
+        arr_data, (_, _, _) = rand_sparse_ndarray(shape,
+                                                  stype,
+                                                  density=density,
+                                                  data_init=data_init,
+                                                  dtype=dtype,
+                                                  modifier_func=modifier_func)
+    else:
+        raise str("Unknown storage type: " + stype)
+    return arr_data
+
+
+def create_sparse_array_zd(shape, stype, density, data_init=None,
```

Comment:
why do you think that this function is for row_sparse only?  rsp_indices is an argument for row_sparse, btw.